filters/encrypt/filter.go

```diff
@@ -0,0 +1,554 @@
+		return fmt.Errorf("%s: unable to get tags from taggable interface: %w", op, err)
+	}
+	for _, pt := range tags {
+		value, err := pointerstructure.Get(t, pt.Pointer)
+		if err != nil {
+			if errors.Is(err, pointerstructure.ErrNotFound) {
+				continue
+			} else {
+				return fmt.Errorf("%s: unable to get value using tag pointer structure (pointer == %s): %w", op, pt.Pointer, err)
+			}
+		}
+		rv := reflect.Indirect(reflect.ValueOf(value))
+		info := &tagInfo{
+			Classification: pt.Classification,
+			Operation:      pt.Filter,
+		}
+		if err = ef.filterValue(ctx, rv, info, withPointer(t, pt.Pointer)); err != nil {
+			return fmt.Errorf("%s: %w", op, err)
+		}
+	}
+	return nil
+}
+
+// filterSlice will filter a slice reflect.Value
+func (ef *Filter) filterSlice(ctx context.Context, classificationTag *tagInfo, slice reflect.Value, opt ...Option) error {
+	const op = "event.(Filter).filterSlice"
+	if classificationTag == nil {
+		return fmt.Errorf("%s: missing classification tag: %w", op, ErrInvalidParameter)
+	}
+	// check for nil value (prevent panics)
+	if slice == reflect.ValueOf(nil) {
+		return nil
+	}
+
+	if slice.Kind() == reflect.Ptr && !slice.IsNil() {
+		slice = slice.Elem()
+	}
+
+	ftype := slice.Type()
+	if ftype != reflect.TypeOf([]string{}) && ftype != reflect.TypeOf([]*string{}) && ftype != reflect.TypeOf([][]uint8{}) {
+		return fmt.Errorf("%s: slice parameter is not a []string or [][]byte: (%s): %w", op, slice.String(), ErrInvalidParameter)
+	}
+	if classificationTag.Classification == PublicClassification {
+		return nil
+	}
+	if slice.Len() == 0 {
+		return nil
+	}
+
+	for i := 0; i < slice.Len(); i++ {
+		if err := ef.filterValue(ctx, slice.Index(i), classificationTag); err != nil {
+			return fmt.Errorf("%s: %w", op, err)
+		}
+	}
+	return nil
+}
+
+// filterValue will filter a value based on it's DataClassification
+func (ef *Filter) filterValue(ctx context.Context, fv reflect.Value, classificationTag *tagInfo, opt ...Option) error {
+	const op = "event.(Filter).filterValue"
+	if classificationTag == nil {
+		return fmt.Errorf("%s: missing classification tag: %w", op, ErrInvalidParameter)
+	}
+
+	// check for nil value (prevent panics)
+	if fv == reflect.ValueOf(nil) {
+		return nil
+	}
+
+	if fv.Kind() == reflect.Ptr && fv.Elem().Kind() == reflect.String {
+		fv = fv.Elem()
+	}
+
+	opts := getOpts(opt...)
+	ftype := fv.Type()
+	if ftype != reflect.TypeOf("") && ftype != reflect.TypeOf([]uint8(nil)) && opts.withPointerstructureInfo == nil {
+		return fmt.Errorf("%s: field value is not a string, []byte or tagged map value: %s :%w", op, fv.String(), ErrInvalidParameter)
+	}
+
+	// check to see if it's an exported struct field
+	if opts.withPointerstructureInfo == nil && !fv.CanSet() {
+		return nil
+	}
+
+	// make sure it's not a []uint8 nil ptr
+	if ftype == reflect.TypeOf([]uint8(nil)) && fv.IsNil() {
+		return nil
+	}
+
+	switch classificationTag.Classification {
+	case PublicClassification:
+		return nil
```

Comment:
I feel like you could save a lot of reflect calls (which are expensive) by moving this check up higher to where you check if the classification tag is nil.

Reply:
ty!